func.go

```diff
@@ -317,6 +343,41 @@ func RegisterFunc(fptr interface{}, cfn uintptr) {
 	fn.Set(v)
 }
 
+func isAllSameFloat(ty reflect.Type) bool {
+	first := ty.Field(0).Type.Kind()
+	if first != reflect.Float32 && first != reflect.Float64 {
+		return false
+	}
+	for i := 0; i < ty.NumField(); i++ {
+		f := ty.Field(i)
+		switch f.Type.Kind() {
+		case first:
+		default:
+			return false
+		}
+	}
+	return true
+}
+
+func areStructFieldsSupported(ty reflect.Type) {
```

Comment:
Sorry I was misunderstanding that this function returned a bool, but this function just does assertions. So `checkStructFieldsSupported` might be better? Sorry for going back and forth.